tests/src/CompilerTest.php

```diff
@@ -767,4 +794,35 @@ private function verify($scenarioJson, $method, $arguments, $mysql, $phpInput, $
             );
         }
     }
+
+    private function verifyException($scenarioJson, $method, $arguments, $code, $data)
+    {
+        // cinnabari stuff
```

Comment:
I should also admit that I've erred on the side of too few comments. I've left out valuable "why" comments in the past. When I was the only one working on the project, I felt that it wasn't necessary. But now that the project is picking up momentum, I wish that there were more "why" comments. I'll try to do better in the future.